runtime/help/plugins.md

```diff
@@ -155,98 +188,117 @@ The packages and functions are listed below (in Go type signatures):
-       type of the value internally.
-* `micro/shell`
-    - `ExecCommand(name string, arg ...string) (string, error)`: runs an
-       executable with the given arguments, and pipes the output (stderr
-       and stdout) of the executable to an internal buffer, which is
-       returned as a string, along with a possible error.
+    - `SetGlobalOptionNative(option string, value interface{}) error`:
+       sets an option to a given value, where the type of [value][EmptyIface] is
+       the actual type of the value internally. Can return an [error][Error] if
+       the provided value is not valid for the given option.
 
-    - `RunCommand(input string) (string, error)`: same as `ExecCommand`,
-       except this uses micro's argument parser to parse the arguments from
-       the input. For example `cat 'hello world.txt' file.txt`, will pass
-       two arguments in the `ExecCommand` argument list (quoting arguments
-       will preserve spaces).
+    - `ConfigDir`:
+        the path to micro's currently active config directory.
 
-    - `RunBackgroundShell(input string) (func() string, error)`: returns a
-       function that will run the given shell command and return its output.
+[Buffer]: https://pkg.go.dev/github.com/zyedidia/micro/v2/internal/buffer#Buffer
+[Completer]: https://pkg.go.dev/github.com/zyedidia/micro/v2/internal/buffer#Completer
+[Error]: https://pkg.go.dev/builtin#error
+[EmptyIface]: https://go.dev/tour/methods/14
+[Match]: https://pkg.go.dev/path/filepath#Match
+
+* `micro/shell`
+    - `ExecCommand(name string, arg ...string) (string, error)`:
+       runs an executable with the given arguments, and pipes the
+       output (stderr and stdout) of the executable to an internal
+       buffer, which is returned as a string, along with a
+       possible [error][Error].
+
+    - `RunCommand(input string) (string, error)`:
+       same as `ExecCommand`, except this uses micro's argument parser to
+       parse the arguments from the input.
+       For example, `cat 'hello world.txt' file.txt` will pass two arguments
+       in the `ExecCommand` argument list (quoting arguments will preserve
+       spaces).
+
+    - `RunBackgroundShell(input string) (func() string, error)`:
+       returns a function that will run the given shell command and return its
+       output.
 
     - `RunInteractiveShell(input string, wait bool, getOutput bool)
                           (string, error)`:
        temporarily closes micro and runs the given command in the terminal.
        If `wait` is true, micro will wait for the user to press enter before
-       returning to text editing. If `getOutput` is true, micro redirect
+       returning to text editing. If `getOutput` is true, micro will redirect
        stdout from the command to the returned string.
 
     - `JobStart(cmd string, onStdout, onStderr,
                 onExit func(string, []interface{}), userargs ...interface{})
-                *exec.Cmd`:
+                *Cmd`:
```

Comment:
Why?

Reply:
After adding links, it seemed to me that the `exec.` (and possibly `buffer.` and `display.`) were just unnecessary noise, but that might just be my perspective. In fact, I think that just dumping go function signatures directly into the plugin docs is not that good of an idea, seeing as plugins are written in Lua. It expects that plugin authors understand at least some subset of go, which is likely not true in a lot of cases.

That said, I decided not to change the layout of all the function signatures because it seemed too big of a change for what I'm trying to achieve here (in retrospect, I should have left the layout alone too and just focused on the content of the docs). I did attempt to do it (for a little while at least), just to explore the possibility space a bit, but I didn't come up with a strictly better layout (in regards to being readable in both micro and in a markdown viewer).

Reply:
> In fact, I think that just dumping go function signatures directly into the plugin docs is not that good of an idea, seeing as plugins are written in Lua.

But this is not just about docs, it is a deeper "issue". The fact is that although plugin authors write their plugins in Lua, for the most part they use APIs that are defined in Go, not in Lua. Just because that's how Micro's plugin system is designed, for good or bad. So the docs should not pretend that it's not like that, I think.

> It expects that plugin authors understand at least some subset of go, which is likely not true in a lot of cases.

But the reality is that they do need to understand at least some subset of go (see above). At least to understand signatures of Buffer or BufPane methods, and so on.